src/consensus/encode.rs

```diff
@@ -545,21 +545,35 @@ impl Encodable for [u16; 8] {
     }
 }
 
+#[inline]
+fn encode_vec<S: io::Write, T: Encodable>(mut s: S, vec: &Vec<T>) -> Result<usize, io::Error> {
+    let mut len = 0;
+    len += VarInt(vec.len() as u64).consensus_encode(&mut s)?;
+    for c in vec.iter() {
+        len += c.consensus_encode(&mut s)?;
+    }
+    Ok(len)
+}
+
 // Vectors
 macro_rules! impl_vec {
     ($type: ty) => {
         impl Encodable for Vec<$type> {
             #[inline]
             fn consensus_encode<S: io::Write>(
                 &self,
-                mut s: S,
+                s: S,
             ) -> Result<usize, io::Error> {
-                let mut len = 0;
-                len += VarInt(self.len() as u64).consensus_encode(&mut s)?;
-                for c in self.iter() {
-                    len += c.consensus_encode(&mut s)?;
-                }
-                Ok(len)
+                encode_vec(s, &self)
```

Comment:
Not sure why we are taking self by reference here in explicit form; omitting `&` should also work (as it assumes reference from the `&self` in method argument)

Reply:
Fixed